manifest: updated Matter SDK revision to pull recent changes (#10736)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

The following list summarizes the most important changes inherited from the upstream Matter:

* Added onboarding code generation to the factory data generation script.

>   Now, using factory data script, users can generate a manual pairing code and QR Code required to commission a Matter-enabled device over Bluetooth LE.

Reviewer: ```suggestion
  You can now use the factory data script to generate a manual pairing code and a QR Code that are required to commission a Matter-enabled device over Bluetooth LE.
```

>   Generated onboarding codes should be put on the device's package or the device itself.

Reviewer: ```suggestion
  Generated onboarding codes should be put on the device's package or on the device itself.
```
